tests/executionspec/spec.rs

```diff
@@ -0,0 +1,321 @@
+                )
+                .unwrap_or(serde_json::Value::Null),
+                text_body: None,
+            };
+
+            let snapshot_name = format!("execution_spec__{}_test_{}", spec.safe_name, i);
+
+            insta::assert_json_snapshot!(snapshot_name, response);
+        }
+
+        mock_http_client.test_hits(&spec.path);
+    }
+}
+
+async fn test_spec(spec: ExecutionSpec) {
+    // insta settings
+    let mut insta_settings = insta::Settings::clone_current();
+    insta_settings.set_prepend_module_to_snapshot(false);
+    let _guard = insta::Settings::bind_to_scope(&insta_settings);
+
+    let mock_http_client = Arc::new(MockHttpClient::new(&spec));
+
+    // check sdl error if any
+    if is_sdl_error(spec.clone(), mock_http_client.clone()).await {
+        return;
+    }
+
+    // Parse and validate all server configs + check for identity
+    let server = check_server_config(spec.clone()).await;
+
+    // merged: Run merged specs
+    let merged = server
+        .iter()
+        .fold(Config::default(), |acc, c| acc.merge_right(c.clone()))
+        .to_sdl();
+
+    let snapshot_name = format!("execution_spec__{}_merged", spec.safe_name);
+
+    insta::assert_snapshot!(snapshot_name, merged);
+
+    // Resolve all configs
+    let mut runtime = runtime::create_runtime(mock_http_client.clone(), spec.env.clone(), None);
+    runtime.file = Arc::new(MockFileSystem::new(spec.clone()));
+    let reader = ConfigReader::init(runtime);
+
+    let server: Vec<ConfigModule> = join_all(
+        server
+            .into_iter()
+            .map(|config| reader.resolve(config, spec.path.parent())),
+    )
+    .await
+    .into_iter()
+    .enumerate()
+    .map(|(i, result)| {
+        result.unwrap_or_else(|e| {
+            panic!(
+                "Couldn't resolve GraphQL in server definition #{} of {:#?}: {}",
+                i + 1,
+                spec.path,
+                e
+            )
+        })
+    })
+    .collect();
+
+    // client: Check if client spec matches snapshot
+    if server.len() == 1 {
+        let config = &server[0];
+
+        let client = print_schema(
+            (Blueprint::try_from(config)
+                .context(format!("file: {}", spec.path.to_str().unwrap()))
+                .unwrap())
+            .to_schema(),
+        );
+        let snapshot_name = format!("execution_spec__{}_client", spec.safe_name);
+
+        insta::assert_snapshot!(snapshot_name, client);
+    }
+
+    // run query tests
+    run_query_tests_on_spec(spec, server, mock_http_client).await;
+}
+
+pub async fn load_and_test_execution_spec(path: &Path) -> anyhow::Result<()> {
+    let contents = fs::read_to_string(path)?;
+    let spec: ExecutionSpec = ExecutionSpec::from_source(path, contents)
+        .await
+        .with_context(|| path.display().to_string())?;
+
+    match spec.runner {
+        Some(Annotation::Skip) => {
+            println!("{} ... {}", spec.path.display(), "skipped".blue());
```

Comment:
looks a little bit off in the output.

```
test run_execution_spec::call-multiple-steps-piping.md                    ... ok
tests/execution/graphql-dataloader-batch-keys.md ... skipped
test run_execution_spec::graphql-dataloader-batch-keys.md                 ... ok
test run_execution_spec::batching-default.md                              ... ok
```

can we improve it?
You can take a look how ok tests are logged in https://github.com/nextest-rs/datatest-stable/blob/main/src/utils.rs#L34 and https://github.com/LukasKalbertodt/libtest-mimic/blob/master/src/printer.rs#L118

Reply:
looking into it.

Reply:
i think We need to track the max length of test file_name

Reply:
if it's complicated let's not bother for now